src/v/cluster/archival/archival_metadata_stm.cc

```diff
@@ -1264,7 +1264,7 @@ archival_metadata_stm::apply_local_snapshot(
     _compacted_replaced_bytes = 0;
 
     if (snap.dirty == state_dirty::dirty) {
-        _last_clean_at = model::offset{0};
+        _last_clean_at = model::offset{};
```

Comment:
This change is better than before but ultimately I believe we need to put this in the local snapshot. If we don't, restarting Redpanda always causes a manifest upload (judging by reading the code), why is that needed? Is this accidental or has meaning?
Second, if we restart and quickly unmount the topic then the finalization logic will assume the remote snapshot does not exist, which is clearly not the case.
So I'm concluding that we need to serialize _last_clean_at together with _last_clean_at to avoid all issues.
Would very much like @Lazin to also take a look.

Reply:
I'm not against serializing `_last_clean_at` and `_last_dirty_at`. But it's difficult to do in a minor release. It should be backward compatible and forward compatible in this case.